### ISSUE TITLE
Redirector: redirect() to DB-sourced URLs fires false-positive TaintedHeader + TaintedSSRF

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#686]
`Redirector::to()` and friends are annotated as both `ssrf` and `header` sinks. This is correct in principle, but in practice the redirect target is almost always derived from a DB model's `url()` method — not directly from user input.

Common false-positive pattern:
```php
$id = $request->input('id');           // tainted
$status = Status::findOrFail($id);     // DB fetch — result should be trusted
return redirect($status->url());       // fires TaintedSSRF + TaintedHeader
```

The taint propagates through `findOrFail()` because Eloquent model-fetch methods don't escape taint. Once data is retrieved from the DB, it's no longer attacker-controlled in the SSRF sense — the attacker can only influence which row is fetched, not the content of `url()`.

Related to #685: adding taint escapes to Eloquent fetch methods would fix this too.

Observed in pixelfed (4 TaintedHeader + 4 TaintedSSRF pairs all from this pattern).
